cloudpai/client.go

```diff
@@ -0,0 +1,117 @@
+package cloudpai
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/vektah/gqlparser/v2/gqlerror"
+
+	"time"
+)
+
+// defaultURL for Atlas Cloud.
+const defaultURL = "https://api.atlasgo.cloud/query"
+
+// roundTripper is a http.RoundTripper that adds the Authorization header.
+type roundTripper struct {
+	token string
+}
+
+// RoundTrip implements http.RoundTripper.
+func (r *roundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	req.Header.Set("Authorization", "Bearer "+r.token)
+	req.Header.Set("User-Agent", "gh-atlas-cli")
+	req.Header.Set("Content-Type", "application/json")
+	return http.DefaultTransport.RoundTrip(req)
+}
+
+// Client is a client for the Atlas Cloud API.
+type Client struct {
+	client   *http.Client
+	endpoint string
+	token    string
```

Comment:
I think the Client does not need the token, does it? You pass it directly to the roundtripper in the constructor.